worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/ObservedDevice.scala

```diff
@@ -19,41 +19,43 @@ package org.apache.celeborn.service.deploy.worker.storage
 
 import java.io.File
 import java.util
-import java.util.{Set => jSet}
+import java.util.{Set => JSet}
 import java.util.concurrent.ConcurrentHashMap
 
 import scala.collection.JavaConverters._
 import scala.io.Source
 
-import org.slf4j.LoggerFactory
-
 import org.apache.celeborn.common.CelebornConf
+import org.apache.celeborn.common.internal.Logging
 import org.apache.celeborn.common.meta.{DeviceInfo, DiskInfo, DiskStatus}
 import org.apache.celeborn.common.metrics.source.AbstractSource
+// Can Remove this if celeborn don't support scala211 in future
+import org.apache.celeborn.common.util.FunctionConverter._
 import org.apache.celeborn.common.util.JavaUtils
 
-class ObservedDevice(val deviceInfo: DeviceInfo, conf: CelebornConf, workerSource: AbstractSource) {
+class ObservedDevice(val deviceInfo: DeviceInfo, conf: CelebornConf, workerSource: AbstractSource)
+  extends Logging {
 
-  val logger = LoggerFactory.getLogger(classOf[ObservedDevice])
+  val diskInfos: ConcurrentHashMap[String, DiskInfo] =
+    deviceInfo.diskInfos.foldLeft(JavaUtils.newConcurrentHashMap[String, DiskInfo]) {
```

Comment:
I'm not sure if ```foldLeft``` is preferred here

Reply:
readable is the most important thing here, considering the developers may not use foldLeft frequently, I reverted to the original style to gain more readability.